src/imageio/imageio_common.h

```diff
@@ -58,6 +58,8 @@ typedef enum dt_imageio_levels_t
   IMAGEIO_CHANNEL_MASK = 0xFF00
 } dt_imageio_levels_t;
 
+// Check that the image is raw by file extension
+gboolean dt_fileext_is_raw(const char *extension);
```

Comment:
For consistency I'll prefer:

`gboolean dt_imageio_is_raw(const char *filename);`

That is, taking a filename and doing the work to extract the extension which we don't want to impose to all callers.

Reply:
I do not agree that the need to get an extension is _imposed_ on callers, it is a _practical necessity_ and below is an explanation why.

__DNG can contain anything__. Therefore, it would be wrong to include this extension in the list of guaranteed raw extensions. We have to consider the possibility of using this function where the caller expects some guarantee that the function will return reliable information that the image in the file is raw.

But in the import dialog filter, it makes sense to allow DNG files in the ignoring non-raw files mode. At least because although ".dng" is not a guarantee of raw, there are cameras where DNG is the native format of raw files: Leica, Pentax, all smartphones.

This means that in the import filter we still _have to extract the extension_ to check it for ".dng". And the extension will also need to be extracted if we decide to use this function to limit calls to the rawspeed and libraw loaders in the imageio image reading workflow (the reason is the same, we have to try to read the dng).

I have to admit, you're absolutely right that I didn't choose a very well-thought-out function name. It is better to name it something like `dt_imageio_is_raw_by_extension` (that is, add the code section name after dt_). I'll change the name of the function.